joomla/joomla-cms
# [4.0] add fields items to contact menu

Pull Request for replace of #23366 .

### Summary of Changes

Add menu items for fields and field groups below the "Contact" menu item in the admin menu.

On update, use script.php to add these new menu items so that lft and rgt are set correctly in the (nested) menu table. 

### Testing Instructions

If you have both MySQL and PostgreSQL, please test on both if possible.

**Test 1: New installation**

1. Apply the patch on a clean 4.0-dev branch using git or merging manually, or apply it on an installation of current 4.0-dev using patchtester.
2. If used patchtester on an existing installation in step 1, delete file `configuration.php` and delete all Joomla database tables in PhpMyAdmin or PhpPgAdmin (depending on your database type).
3. Do a new installation, login to backend and check the available items below the "Contact" item in the admin menu.

Result: See section "Expected result" below.

**Test 2: Update**

1. Update a 3.9.12 or staging to current 4.0-dev plus the patch of this PR applied by using the "Upload & Update" tab of the Joomla Update Component and the update zip container previously downloaded from following URL: [https://test5.richard-fath.de/Joomla_4.0.0-beta1-dev-Development-Update_Package_2019-10-19_pr-26540.zip](https://test5.richard-fath.de/Joomla_4.0.0-beta1-dev-Development-Update_Package_2019-10-19_pr-26540.zip "https://test5.richard-fath.de/Joomla_4.0.0-beta1-dev-Development-Update_Package_2019-10-19_pr-26540.zip"). ATTENTION: When working on a git repository you should know how to clean up your branch afte this test.
2. Login to backend and check the available items below the "Contact" item in the admin menu.

Result: See section "Expected result" below.

### Expected result

you have:

- Fields
- Fieldsgroup 

below Contact menu item
![Screenshot from 2019-10-10 19-39-04](https://user-images.githubusercontent.com/181681/66592597-aa389480-eb95-11e9-9ad0-9fd582541a45.png)

### Actual result

You don't have these 2 menu items below Contact component

### Additional comments

We need to review how we deal with changes of content in nested tables when updating because issues with that are is quite hard to fix with update sql scripts. (personal consideration)
